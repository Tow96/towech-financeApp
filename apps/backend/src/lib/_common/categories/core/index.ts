export * from './category-aggregate';
export * from './category-events';
export * from './i-category-repository';
export * from './subcategory-entity';
export * from './subcategory-events';
