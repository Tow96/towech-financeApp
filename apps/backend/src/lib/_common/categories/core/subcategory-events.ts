// Slice packages
import { CategoryType } from './category-aggregate';

// Internal references
import { DomainEvent, DomainEventProps } from '../../primitives';

// ----------------------------------------------
export class SubCategoryCreatedEvent extends DomainEvent {
  readonly userId: string;
  readonly categoryId: string;
  readonly name: string;
  readonly type: CategoryType;

  constructor(props: DomainEventProps<SubCategoryCreatedEvent>) {
    super(props);
    this.userId = props.userId;
    this.categoryId = props.categoryId;
    this.name = props.name;
    this.type = props.type;
  }
}

export class SubCategoryUpdatedEvent extends DomainEvent {
  readonly categoryId: string;
  readonly type: CategoryType;
  readonly name: string;

  constructor(props: DomainEventProps<SubCategoryUpdatedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
    this.type = props.type;
    this.name = props.name;
  }
}

export class SubCategoryRemovedEvent extends DomainEvent {
  readonly categoryId: string;

  constructor(props: DomainEventProps<SubCategoryRemovedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
  }
}
