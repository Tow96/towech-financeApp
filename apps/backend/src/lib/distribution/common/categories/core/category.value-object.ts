// App packages
import { ValueObject } from '../../../../_common/primitives/value-object.base';

export enum CategoryType {
  income = 'INCOME',
  expense = 'EXPENSE',
  transfer = 'TRANSFER',
}

interface CategoryProps {
  _id: string;
  _userId: string;
  _type: CategoryType;
}

export class Category extends ValueObject<CategoryProps> {
  get id() {
    return this.props._id;
  }

  get userId() {
    return this.props._userId;
  }

  get type() {
    return this.props._type;
  }
}
