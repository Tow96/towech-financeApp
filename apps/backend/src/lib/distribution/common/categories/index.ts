export * from './core/category.value-object';
