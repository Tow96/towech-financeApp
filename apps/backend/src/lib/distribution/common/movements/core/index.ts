export * from './category.value-object';
export * from './i-movement-repository';
export * from './movement-aggregate';
export * from './summary-item.value-object';
export * as events from './movement-events';
