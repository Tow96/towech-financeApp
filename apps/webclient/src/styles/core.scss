// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}
