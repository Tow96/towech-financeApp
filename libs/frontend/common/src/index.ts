import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';

export * from './lib/ConditionalClasses';
export * from './lib/Tanstack.Provider';
export * from './lib/Components';
export * from './lib/Http.Context';

library.add(fas);
