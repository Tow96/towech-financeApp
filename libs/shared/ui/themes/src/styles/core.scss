// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
