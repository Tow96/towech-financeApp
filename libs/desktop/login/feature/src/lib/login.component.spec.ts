// Libraries
import { Component, EventEmitter } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
import { AsyncPipe } from '@angular/common';
// import { subscribeSpyTo } from '@hirez_io/observer-spy';
// Tested elements
import { PatchFormGroupValuesDirective } from './login.component';
// Mocks
// import { provideStore } from '@ngrx/store';
// import { adaptReducer } from '@state-adapt/core';
// import { DesktopUserService } from '@towech-finance/desktop/user/data-access';
// import { LoginUser } from '@towech-finance/shared/utils/models';

@Component({
  standalone: true,
  imports: [PatchFormGroupValuesDirective, ReactiveFormsModule, AsyncPipe],
  template: ` <form [formGroup]="form" [patchFormGroupValues]="patcher$ | async"></form> `,
})
class TestComponent {
  public form = new FormGroup({
    user: new FormControl(''),
    pass: new FormControl(''),
  });

  public patch = new EventEmitter<{ user: string; pass: string }>();
  public patcher$ = this.patch.pipe();
}

describe('Patch form Directive', () => {
  let component: TestComponent;
  let fixture: ComponentFixture<TestComponent>;
  let spy: jest.SpyInstance;

  beforeEach(() => {
    jest.clearAllMocks();
    TestBed.configureTestingModule({ imports: [TestComponent] });
    fixture = TestBed.createComponent(TestComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    spy = jest.spyOn(component.form, 'patchValue');
  });

  describe('When the patch pipe is called without anything', () => {
    it('Should not modify the form', () => {
      const prevValue = { ...component.form.value };
      // component.patcher$.next(undefined);
      const newValue = { ...component.form.value };

      expect(prevValue).toEqual(newValue);
      expect(spy).toHaveBeenCalledTimes(0);
    });
  });

  describe('When the patch pipe is called with an update', () => {
    it('Should patch the form and have it NOT emit', () => {
      const newValue = { user: 'newuser', pass: 'password' };
      component.patch.next(newValue);
      fixture.detectChanges();
      expect(component.form.value).toEqual(newValue);
      expect(spy).toHaveBeenCalledWith(newValue, { emitEvent: false });
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});

describe('Desktop Login Component', () => {
  // const newValue: LoginUser = { username: 'user', password: 'pass', keepSession: false };
  // let component: DesktopLoginComponent;
  // let fixture: ComponentFixture<DesktopLoginComponent>;
  // let service: DesktopUserService;
  // let compiled: HTMLElement;

  it('', () => {
    expect(1).toBeTruthy();
  });
  // beforeEach(() => {
  //   jest.clearAllMocks();
  //   TestBed.configureTestingModule({
  //     providers: [provideStore({ adapt: adaptReducer }), DesktopUserService],
  //   });
  //   fixture = TestBed.createComponent(DesktopLoginComponent);

  //   component = fixture.componentInstance;
  //   fixture.detectChanges();
  //   compiled = fixture.nativeElement;

  //   service = TestBed.inject<DesktopUserService>(DesktopUserService);
  // });

  // it('Should be defined', () => expect(component).toBeTruthy());

  // it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());

  // describe('When the form fields are changed', () => {
  //   it('Should update the store', () => {
  //     const spy = subscribeSpyTo(component.store.state$);
  //     component.form.patchValue(newValue);

  //     expect(spy.getLastValue()).toEqual(newValue);
  //   });
  // });

  // // TODO: Add form validation before sending and error processing
  // describe('When the login button is pressed', () => {
  //   it('Should next the userservice', () => {
  //     const userSpy = subscribeSpyTo(service.login$);
  //     const loginBttn: HTMLElement =
  //       fixture.debugElement.nativeElement.querySelector('#Login-button > button');
  //     loginBttn.click();
  //     expect(userSpy.receivedNext()).toBe(true);
  //   });
  // });
});
