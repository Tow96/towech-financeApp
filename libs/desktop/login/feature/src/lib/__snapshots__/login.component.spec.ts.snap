// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Login Component Must match the snapshot 1`] = `
<div
  id="root1"
>
  <towech-finance-toaster>
    <div
      class="desktop-toaster"
    >
      
    </div>
  </towech-finance-toaster>
  <div
    class="login-container"
  >
    <h1>
      Login
    </h1>
    <towech-finance-login-form>
      <form
        class="ngrx-forms-invalid ngrx-forms-pristine ngrx-forms-untouched ngrx-forms-unsubmitted"
        novalidate=""
      >
        <towech-finance-shared-input
          class="ngrx-forms-invalid ngrx-forms-pristine ngrx-forms-untouched ngrx-forms-unsubmitted"
          label="Username"
        >
          <div
            class="input_component"
          >
            <input
              placeholder=" "
              type="text"
            />
            <label>
              Username
            </label>
          </div>
        </towech-finance-shared-input>
        <towech-finance-shared-input
          class="ngrx-forms-invalid ngrx-forms-pristine ngrx-forms-untouched ngrx-forms-unsubmitted"
          label="Password"
          type="password"
        >
          <div
            class="input_component"
          >
            <input
              placeholder=" "
              type="password"
            />
            <label>
              Password
            </label>
          </div>
        </towech-finance-shared-input>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </towech-finance-login-form>
  </div>
  
</div>
`;
